### PR TITLE
Japanese text conversion input

### PR DESCRIPTION
**Pull request by mitsuyapega**
Add an input with the following 3 features. All of them convert characters with onBlur.

* Lowercase to Uppercase
Converts Lowercase to Uppercase, for when the client inputs lowercase and the internal character handles it as uppercase.
* Hiragana to Katakana
Japanese has two types of characters, Hiragana and Katakana, which correspond one-to-one. Hiragana input is converted to Katakana.
* Full-width to Half-width
In Japanese, characters are generally handled as multi-byte full-width characters rather than single-byte half-width characters, and for this reason Japanese keyboards have a function to input characters in full-width. Some full-width characters have the same meaning in half-width, and these are converted from full-width to half-width.

### CODE REVIEW

**Comment by ricmars** [comment]
please clean up - lots of files are not necessary - follow the best practices and look at other components

**Comment by ricmars** [comment]
overall - you should remove the other components like password input or mask input and follow the same practice

File: src/components/Pega_Extensions_JapaneseInput/index.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
+import { useEffect, useState, useRef } from 'react';
+import { Input, FieldValueList, Text, withConfiguration } from '@pega/cosmos-react-core';
+import '../create-nonce';
+
+// interface for props
+interface PegaExtensionsJapaneseInputProps {
+  // If any, enter additional props that only exist on TextInput here
+  hasSuggestions?: boolean;
+  variant?: any;
```
Comment (Sujesh95) [inline src/components/Pega_Extensions_JapaneseInput/index.tsx:9]: Provide appropriate types in the place of ```any```

File: src/components/Pega_Extensions_JapaneseInput/index.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
+import { useEffect, useState, useRef } from 'react';
+import { Input, FieldValueList, Text, withConfiguration } from '@pega/cosmos-react-core';
+import '../create-nonce';
+
+// interface for props
+interface PegaExtensionsJapaneseInputProps {
```
Comment (Sujesh95) [inline src/components/Pega_Extensions_JapaneseInput/index.tsx:6]: export the interface incase we need it.

Comment (Sujesh95) [inline src/components/Pega_Extensions_JapaneseInput/index.tsx:6]: extend InputProps from @pega/cosmos-react-core. In that case, you can remove duplicate types.

**Review by ricmars — APPROVED** [review]

**Pull request merged**